pkg/splunk/enterprise/workerpool.go

```diff
@@ -127,29 +128,32 @@ func (w *AppInstallWorkerPool) Start() {
 
 					// add the job to the jobs channel
 					w.JobsChan <- app
-					jobsAdded++
+					jobsParsed++
+					actualJobsAdded = true
 				} else {
 					_ = fmt.Errorf("not enough space to accomodate app:%s, availableDiskSpace:%d", app.AppName, w.availableDiskSpace)
 					continue
 				}
 			case enterpriseApi.DownloadComplete:
 				// This is done in case if all the apps were already downloaded, we would need to
 				// break from outer loop and close the channel as there is no job to add.
-				jobsAdded++
+				jobsParsed++
 				continue
 			default:
 				continue
 			}
 		}
 
 		// all the jobs are added to the channel, we are done here
-		if jobsAdded == len(w.JobList) {
+		if jobsParsed == len(w.JobList) {
```

Comment:
So if all jobs are in `DownloadInProgress` or `DownloadError` we will never reach this condition, correct?

Reply:
So `DownloadInProgress` is just a transient state before the actual download begins. If due to some reason download is taking more time, we will loop through and will increment the jobsParsed count once the download is complete.
But, about the `DownloadError` state, I think you are right. We probably have to join `DownloadComplete` and `DownloadError` in a single case. Let me check and fix it.